src/test_l1_dsHdmiIn.c

```diff
@@ -2391,67 +2341,75 @@ void test_l1_dsHdmiIn_negative_dsGetAVLatency_sink(void) {
- * 
+ *
  * **Test Procedure:**@n
  * |Variation / Step|Description|Test Data|Expected Result|Notes|
  * |:--:|-----------|----------|--------------|-----|
  * |01|Initialize HDMI input using dsHdmiInInit() | | dsERR_NONE | Should Pass |
- * |02|Get the number of input ports from the profile file
- * |03|Call dsSetEdid2AllmSupport() with all valid ports and enable EDID ALLM support | iHdmiPort: dsHDMI_IN_PORT_0, allmSupport: true | dsERR_NONE | Should Pass |
- * |04|Call dsSetEdid2AllmSupport() to disable EDID ALLM support | iHdmiPort: dsHDMI_IN_PORT_0, allmSupport: false | dsERR_NONE | Should Pass |
- * |05|Call dsSetEdid2AllmSupport() with valid ports | dsERR_OPERATION_NOT_SUPPORTED | For source devices
- * |06|Terminate the HDMI input using dsHdmiInTerm() | | dsERR_NONE | Clean up after test |
- * 
+ * |02|Call dsSetEdid2AllmSupport() with all valid ports and enable EDID ALLM support | iHdmiPort: dsHDMI_IN_PORT_0, allmSupport: true | dsERR_NONE | Should Pass |
+ * |03|Call dsSetEdid2AllmSupport() to disable EDID ALLM support | iHdmiPort: dsHDMI_IN_PORT_0, allmSupport: false | dsERR_NONE | Should Pass |
+ * |04|Call dsSetEdid2AllmSupport() with valid ports | dsERR_OPERATION_NOT_SUPPORTED | For source devices
+ * |05|Terminate the HDMI input using dsHdmiInTerm() | | dsERR_NONE | Clean up after test |
+ *
  */
 void test_l1_dsHdmiIn_positive_dsSetEdid2AllmSupport_sink(void) {
+
     gTestID = 47;
     UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t  result = dsERR_NONE;
     uint8_t numInputPorts = 0;
+    numInputPorts = UT_KVP_PROFILE_GET_UINT8("dsHdmiIn/HdmiInputPort/numberOfPorts");
 
     // Step 1: Initialize HDMI input using dsHdmiInInit()
     UT_ASSERT_EQUAL_FATAL(dsHdmiInInit(), dsERR_NONE);
 
-    // Step 2: Get the number of input ports from the profile file
-    numInputPorts = UT_KVP_PROFILE_GET_UINT8("dsHdmiIn/HdmiInputPort/numberOfPorts");
-    if (numInputPorts > 0) {
-        // Step 3: Call dsSetEdid2AllmSupport() with all valid ports and enable EDID ALLM support
-        UT_ASSERT_EQUAL(dsSetEdid2AllmSupport(dsHDMI_IN_PORT_0, true), dsERR_NONE);
-        UT_ASSERT_EQUAL(dsSetEdid2AllmSupport(dsHDMI_IN_PORT_1, true), dsERR_NONE);
-        UT_ASSERT_EQUAL(dsSetEdid2AllmSupport(dsHDMI_IN_PORT_2, true), dsERR_NONE);
-    
-        // Step 4: Call dsSetEdid2AllmSupport() to disable EDID ALLM support
-        UT_ASSERT_EQUAL(dsSetEdid2AllmSupport(dsHDMI_IN_PORT_0, false), dsERR_NONE);
-    } else {
-        // Step 5:Call dsSetEdid2AllmSupport() with valid ports
-	UT_ASSERT_EQUAL(dsSetEdid2AllmSupport(dsHDMI_IN_PORT_0, true), dsERR_OPERATION_NOT_SUPPORTED);
+    if (gSourceType == 0) {
+        // Step 2: Call dsSetEdid2AllmSupport() with all valid ports and enable EDID ALLM support
+        for (int port = dsHDMI_IN_PORT_0; port < numInputPorts; port++)
+        {
+              result = dsSetEdid2AllmSupport(port, true);
+              UT_ASSERT_EQUAL(result, dsERR_NONE);
+        }
```

Comment:
for (int port = dsHDMI_IN_PORT_0; port < numInputPorts; port++) {
    // Enable EDID ALLM support
    result = dsSetEdid2AllmSupport(port, true);
    UT_ASSERT_EQUAL(result, dsERR_NONE);
    // Disable EDID ALLM support
    result = dsSetEdid2AllmSupport(port, false);
    UT_ASSERT_EQUAL(result, dsERR_NONE);
}


make this as single loop

Reply:
Done